gqcp/include/QCMethod/HF/UHF/UHFSCFSolver.hpp

```diff
@@ -69,10 +69,29 @@ class UHFSCFSolver {
             .add(UHFElectronicEnergyCalculation<Scalar>());
 
         // Create a compound convergence criterion on the norm of subsequent alpha- and beta-density matrices
-        using SingleConvergenceType = ConsecutiveIteratesNormConvergence<SpinResolvedOneDM<Scalar>, UHFSCFEnvironment<Scalar>>;
+        using SingleConvergenceType = ConsecutiveIteratesNormConvergence<OneDM<Scalar>, UHFSCFEnvironment<Scalar>>;
 
-        const auto density_matrix_extractor = [](const UHFSCFEnvironment<Scalar>& environment) { return environment.density_matrices; };
-        const SingleConvergenceType convergence_criterion {threshold, density_matrix_extractor, "the UHF density matrices in AO basis"};
+        const auto density_matrix_alpha_extractor = [](const UHFSCFEnvironment<Scalar>& environment) {
+            std::deque<GQCP::OneDM<Scalar>> alpha_density_matrices;
+
+            for (size_t i = 0; i < environment.density_matrices.size(); i++) {
+                alpha_density_matrices.push_back(environment.density_matrices[i].alpha());
+            }
+            return alpha_density_matrices;
+        };
+
+        const auto density_matrix_beta_extractor = [](const UHFSCFEnvironment<Scalar>& environment) {
+            std::deque<GQCP::OneDM<Scalar>> beta_density_matrices;
+
+            for (size_t i = 0; i < environment.density_matrices.size(); i++) {
+                beta_density_matrices.push_back(environment.density_matrices[i].beta());
+            }
+            return beta_density_matrices;
+        };
+
+        const SingleConvergenceType convergence_criterion_alpha {threshold, density_matrix_alpha_extractor, "the UHF alpha-density matrix in AO basis"};
+        const SingleConvergenceType convergence_criterion_beta {threshold, density_matrix_beta_extractor, "the UHF beta-density matrix in AO basis"};
+        const CompoundConvergenceCriterion<UHFSCFEnvironment<Scalar>> convergence_criterion {convergence_criterion_alpha, convergence_criterion_beta};
```

Comment:
I understand that this code makes the UHF algorithms re-converge. However, I'll take a little time to check why the convergence didn't work when using the spin-blocked (M x M) density matrix.